lib/types/local-date.js

```diff
@@ -0,0 +1,200 @@
+ * </p>
+ * @param {Number} year The year.
+ * @param {Number} month Between 1 and 12 inclusive.
+ * @param {Number} day Between 1 and the number of days in the given month of the given year.
+ * @constructor
+ */
+function LocalDate(year, month, day) {
+  //implementation detail: internally uses a UTC based date
+  if (typeof year === 'number' && typeof month === 'number' && typeof day === 'number') {
+    this.value = new Date(Date.UTC(year, month - 1, day));
+  }
+  else if (typeof month === 'undefined' && typeof day === 'undefined') {
+    if (typeof year === 'number') {
+      //in milliseconds since unix epoch
+      this.value = new Date(year);
+    }
+  }
+  if (!this.value) {
+    throw new Error('You must provide a valid year, month and day');
+  }
+  var self = this;
+  /**
+   * A number representing the year.
+   * @name year
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  /**
+   * A number between 1 and 12 inclusive representing the month.
+   * @name month
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  /**
+   * A number between 1 and the number of days in the given month of the given year (28, 29, 30, 31).
+   * @name day
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  Object.defineProperties(this, {
+    'year': { enumerable: true, get: function () {
+      return self.value.getUTCFullYear();
+    }},
+    'month': { enumerable: true, get: function () {
+      return self.value.getMonth() + 1;
+    }},
+    'day': { enumerable: true, get: function () {
+      return self.value.getDate();
+    }}
+  });
+}
+
+/**
+ * Creates a new instance of LocalDate using the current year, month and day from the system clock in the default time-zone.
+ */
+LocalDate.now = function () {
+  return LocalDate.fromDate(new Date());
+};
+
+/**
+ * Creates a new instance of LocalDate using the current date from the system clock at UTC.
+ */
+LocalDate.utcNow = function () {
+  //noinspection JSCheckFunctionSignatures
+  return new LocalDate(Date.now());
+};
+
+
+/**
+ * Creates a new instance of LocalDate using the year, month and day from the provided local date time.
+ * @param {Date} date
+ */
+LocalDate.fromDate = function (date) {
+  if (isNaN(date)) {
+    throw new TypeError('Invalid date: ' + date);
+  }
+  return new LocalDate(date.getFullYear(), date.getMonth() + 1, date.getDate());
+};
+
+/**
+ * Creates a new instance of LocalDate using the year, month and day provided in the form: yyyy-mm-dd.
+ * @param {String} value
+ */
+LocalDate.fromString = function (value) {
+  var multiplier = 1;
+  if (value[0] === '-') {
+    value = value.substring(1);
+    multiplier = -1;
+  }
+  var parts = value.split('-');
+  return new LocalDate(multiplier * parseInt(parts[0], 10), parseInt(parts[1], 10), parseInt(parts[2], 10));
```

Comment:
Do you think it would be worth making the fromString implementations of LocalDate and LocalTime have more stricter validation of integer parsing?  parseInt will simply ignore non-numerical values, so you can create some unexpected behavior if non-complete numeric values are provided.